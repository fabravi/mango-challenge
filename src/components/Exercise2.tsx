"use client";
import Range from "@/ui/Range";

import { useState } from "react";

interface Exercise2Props {
  marks: number[];
}

export default function Exercise2({ marks }: Exercise2Props) {
  const min = marks[0];
  const max = marks.at(-1)!;

  const [value, setValue] = useState({ min, max });

  return (
    <div>
      <Range
        min={min}
        max={max}
        marks={marks}
        value={value}
        onChange={setValue}
      />
    </div>
  );
}
