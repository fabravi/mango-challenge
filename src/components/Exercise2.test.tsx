import { render } from "@testing-library/react";
import Exercise2 from "./Exercise2";

describe("Exercise2", () => {
  const marks = [10, 20, 30, 40, 50];

  test("should render correctly", () => {
    const { container } = render(<Exercise2 marks={marks} />);
    expect(container).toMatchSnapshot();
  });

  test("should render the range input", () => {
    const { container } = render(<Exercise2 marks={marks} />);
    const input = container.querySelector("input");
    expect(input).toHaveAttribute("type", "number");
  });

  test("should render the range track", () => {
    const { container } = render(<Exercise2 marks={marks} />);
    const track = container.querySelector("span.track");
    expect(track).toBeInTheDocument();
  });

  test("should render the range mark", () => {
    const { container } = render(<Exercise2 marks={marks} />);
    const mark = container.querySelectorAll("span.mark");
    expect(mark).toHaveLength(5);
  });

  test("should render the range rail", () => {
    const { container } = render(<Exercise2 marks={marks} />);
    const rail = container.querySelector("span.rail");
    expect(rail).toBeInTheDocument();
  });

  test("should render the thumbs", () => {
    const { queryAllByRole } = render(<Exercise2 marks={marks} />);
    const thumbs = queryAllByRole("slider-thumb");
    expect(thumbs).toHaveLength(2);
  });
});
