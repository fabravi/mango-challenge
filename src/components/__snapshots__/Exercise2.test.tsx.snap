// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise2 should render correctly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <fieldset
        class="fieldset"
      >
        <label
          class=""
          for="min"
        >
          €
        </label>
        <input
          class="input "
          data-testid="min"
          max="50"
          min="10"
          readonly=""
          tabindex="-1"
          type="number"
          value="10.00"
        />
      </fieldset>
      <div
        class="range"
      >
        <span
          class="rail"
        />
        <span
          class="track"
          role="slider-track"
          style="left: 0%; width: 100%;"
        />
        <span
          class="thumb "
          role="slider-thumb"
          style="left: 0%;"
          tabindex="0"
        />
        <span
          class="thumb "
          role="slider-thumb"
          style="left: 100%;"
          tabindex="0"
        />
        <span
          class="mark"
          style="left: 0%;"
        />
        <span
          class="mark"
          style="left: 25%;"
        />
        <span
          class="mark"
          style="left: 50%;"
        />
        <span
          class="mark"
          style="left: 75%;"
        />
        <span
          class="mark"
          style="left: 100%;"
        />
      </div>
      <fieldset
        class="fieldset"
      >
        <label
          class=""
          for="max"
        >
          €
        </label>
        <input
          class="input "
          data-testid="max"
          max="50"
          min="10"
          readonly=""
          tabindex="-1"
          type="number"
          value="50.00"
        />
      </fieldset>
    </div>
  </div>
</div>
`;
